server/async_tcp.go

```diff
@@ -80,7 +80,27 @@ func RunAsyncTCPServer(wg *sync.WaitGroup) error {
 		Port: config.Port,
 		Addr: [4]byte{ip4[0], ip4[1], ip4[2], ip4[3]},
 	}); err != nil {
-		return err
+
```

Comment:
The first and foremost fix is logging if errored out from ``bind()``
```go
if err = syscall.Bind(serverFD, &syscall.SockaddrInet4{
		Port: config.Port,
		Addr: [4]byte{ip4[0], ip4[1], ip4[2], ip4[3]},
	}); err != nil {
		log.Fatal(err)
		return err
	}
```
The second part of the fix should be a port scan on a designated range...PORT KNOCKING.

Reply:
ok I think I got this wrong may be. I thought this would be needed only in development mode only in case of port conflict. So this only would solve it in debug mode. Let me rework on this.

- If error from bind, pick up the next free port and try bind right.

Reply:
You don't have to try binding on each port this will result in spaghetti code. You have to create a distant utility to do a port scan on a fixed range. If the scan is affirmative then you can assign the port right away. 

The idea behind the separation of concerns is that my network stack (custom syscall-based server) should be quick and not create garbage for runtime, the more you iterate on bind more useless runtime stack you would create.

JUST AVOID THAT
The process should be blazing fast to startup, it's not complicated enough at this stage of development as well.